perma_web/perma/views/user_management.py

```diff
@@ -1901,39 +1905,62 @@ def sign_up_firm(request):
         if something_took_the_bait := check_honeypot(request, 'register_email_instructions', check_js=True):
             return something_took_the_bait
 
-        form = CreateUserFormWithFirm(request.POST)
+        user_form = CreateUserFormWithFirm(request.POST)
         user_email = request.POST.get('e-address', '').lower()
 
         try:
-            target_user = LinkUser.objects.get(email=user_email)
+            existing_user = LinkUser.objects.get(email=user_email)
         except LinkUser.DoesNotExist:
-            target_user = None
-        if target_user:
-            requested_account_note = request.POST.get('requested_account_note', None)
-            target_user.requested_account_type = 'firm'
-            target_user.requested_account_note = requested_account_note
-            target_user.save()
-            email_firm_request(request, target_user)
+            existing_user = None
+
+        # If user email in form matches an existing user in database, update user record to include
+        # organization name under `LinkUser.requested_account_note` field
+        if existing_user is not None:
+            organization_name = request.POST.get('name', None)
+            existing_user.requested_account_type = 'firm'
+            existing_user.requested_account_note = organization_name
+            existing_user.save()
+            email_firm_request(request, existing_user)
             return HttpResponseRedirect(reverse('firm_request_response'))
```

Comment:
`firm_request_response` renders `registration/firm_request.html`, which looks like this:
![image](https://github.com/user-attachments/assets/7988d521-627c-4be8-9def-3f3b7cfc3712)

I bet we want to switch the wording from firm -> organization.

Reply:
Good catch! Fixed in [c1a6c7b](https://github.com/harvard-lil/perma/pull/3595/commits/c1a6c7beb0a6354cd7adbb58abde79b5ba662046).